src/pages/students/students.scheme.ts

```diff
@@ -0,0 +1,42 @@
+import * as yup from "yup";
+
+export const studentsScheme = yup
+  .object({
+    name: yup.string().required("Debe ingresar el nombre"),
+    lastname: yup.string().required("Debe ingresar el apellido"),
+    nroTelefono: yup
+      .string()
+      .min(8, "Debe tener al menos 8 caracteres")
+      .max(15, "No puede superar los 15 caracteres")
+      .matches(/^[+0-9\s-]+$/, "Solo se permiten números"),
+    fechaNacimiento: yup
+      .date()
+      .nullable()
+      .required("La fecha de nacimiento es obligatoria")
+      .max(
+        new Date(new Date().setFullYear(new Date().getFullYear() - 10)),
+        "Debe tener al menos 10 años"
+      ),
+    tipoPago: yup.string().required("Debe seleccionar un tipo de pago"),
+    cantidadDias: yup
+      .number()
+      .typeError("Debe ingresar un número válido")
+      .required("Debe ingresar la cantidad de días")
+      .min(1, "Debe ser al menos 1 día")
+      .max(7, "No puede superar los 7 días"),
+    fechaIngreso: yup
+      .date()
+      .nullable()
+      .required("La fecha de ingreso es obligatoria"),
+    diaPago: yup.string().when("tipoPago", {
+      is: (val: string) => val === "Día específico",
+      then: (schema) => schema.required("Debe ingresar el día de pago"),
+      otherwise: (schema) => schema.notRequired(),
+    }),
+    diasExtra: yup.string().when("tipoPago", {
+      is: (val: string) => val === "Del 1 al 10",
+      then: (schema) => schema.required("Debe ingresar los días extras"),
+      otherwise: (schema) => schema.notRequired(),
+    }),
```

Comment:
Muy buen con las validaciones estas :) Lo unico que cambiaría es que en vez de verificar el valor contra valores específicos "Día específico" o "Del 1 al 10", llevaria esos valores a un archivo de constantes, porque si el día de mañana tenemos que modificar cómo se muestra el texto en el selector, vas a tener que modificar la validación acá también y para eso siempre es mejor tener en un lugar el valor de las constantes y modificarlas en un solo lugar.

Por ejemplo, podemos definir el siguiente enum

```
export enum PaymentType {
  DIA_ESPECIFICO = "Dia específico",
  PRINCIPIO_MES = "Del 1 al 10"
}
```

y aca tendriamos

```suggestion
    diaPago: yup.string().when("tipoPago", {
      is: (val: string) => val === PaymentType.DIA_ESPECIFICO,
      then: (schema) => schema.required("Debe ingresar el día de pago"),
      otherwise: (schema) => schema.notRequired(),
    }),
    diasExtra: yup.string().when("tipoPago", {
      is: (val: string) => val === PaymentType.PRINCIPIO_MES,
      then: (schema) => schema.required("Debe ingresar los días extras"),
      otherwise: (schema) => schema.notRequired(),
    }),
```

Reply:
Para poder usar enum tuve que comentar esta linea del archivo tsconfig.app.json:
![image](https://github.com/user-attachments/assets/2bab6634-4f14-4eea-a42e-ed9c372e9889)
si no comento esa línea no me deja usarlo y me sale error. Está bien o que tendría que hacer sino?

Reply:
no lo comentes, dejalo en "false"